ls1intum/Artemis
# `Quiz exercises`: Fix text overflow in quiz questions

### Checklist
#### General


#### Client


### Motivation and Context

This change is required because in the case of a very long string that cannot be broken up, the string overflows outside the card component. By adding the `overflow: auto` property to the scss file, the card component will become automatically scrollable. The card component however is only scrollable when necessary. If the string can be broken down onto multiple lines, then the text will wrap rather than be scrollable. Overall this fixes an edge case.

This pull request fixes #4721 

### Description
When inputting a very long string that cannot be broken up into a quiz question exercise, such as `"hsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfasdfasdfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsa"`, this string overflows past the boundaries of the card component.

The text will no longer overflow, instead a scroll bar will appear if and only if the text would overflow otherwise. Currently the text will wrap when possible. However in certain cases when that string cannot be broken up, the text will overflow past the boundaries. In that case, the card will become scrollable.




We made changes to 
the following two files: short-answer-question-component.scss and multiple-choice-question-component.scss

### Steps for Testing

Prerequisites:
-  There shouldn't be any prerequisites but I was logged in as admin to create the quiz

1. Log in to Artemis
2. Navigate to Course Management
3. Click on the "exercises" button
4. Create a quiz exercise and either create a multiple choice question or short answer question.
5. Input the string `"hsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfasdfasdfsahsakjdflahsdfjhasdhadfsahsakjdflahsdfjhasdhadfsa"` on the first line
6. Observe that the string overflows past the card boundaries.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|

<img width="1328" alt="Screen Shot 2022-05-08 at 12 45 53 PM" src="https://user-images.githubusercontent.com/44660375/167313151-042da6b6-711b-4f83-95d5-9562bed95ff4.png">


### Screenshots
Before
<img width="869" alt="Screen Shot 2022-05-08 at 12 29 12 PM" src="https://user-images.githubusercontent.com/44660375/167312547-1e2510c0-f2ef-41a7-8e99-94b9cf412af9.png">

After
<img width="886" alt="Screen Shot 2022-05-08 at 12 32 03 PM" src="https://user-images.githubusercontent.com/44660375/167312656-4a3c4564-cd7f-4141-a6c8-97575ddc62dc.png">
